cscs-checks/apps/pytorch/pytorch_horovod_check.py

```diff
@@ -0,0 +1,98 @@
+# Copyright 2016-2020 Swiss National Supercomputing Centre (CSCS/ETH Zurich)
+# ReFrame Project Developers. See the top-level LICENSE file for details.
+#
+# SPDX-License-Identifier: BSD-3-Clause
+
+import reframe as rfm
+import reframe.utility.sanity as sn
+import reframe.utility.osext as osext
+
+
+@rfm.parameterized_test(*[[model, mpi_task]
+                          for mpi_task in [32, 8, 2, 1]
+                          for model in ['inception_v3', 'resnet50']])
+class PytorchHorovodTest(rfm.RunOnlyRegressionTest):
+    def __init__(self, model, mpi_task):
+        self.descr = 'Distributed training with Pytorch and Horovod'
+        self.valid_systems = ['daint:gpu']
+        if mpi_task < 20:
+            self.valid_systems += ['dom:gpu']
+
+        self.valid_prog_environs = ['builtin']
+        cray_cdt_version = osext.cray_cdt_version()
+        self.modules = [f'Horovod/0.19.5-CrayGNU-{cray_cdt_version}-pt-1.6.0']
+        self.num_tasks_per_node = 1
+        self.num_cpus_per_task = 12
+        self.num_tasks = mpi_task
+        batch_size = 64
+        self.variables = {
+            'NCCL_DEBUG': 'INFO',
+            'NCCL_IB_HCA': 'ipogif0',
+            'NCCL_IB_CUDA_SUPPORT': '1',
+            'OMP_NUM_THREADS': '$SLURM_CPUS_PER_TASK',
+        }
+        hash = 'master'
+        git_url = f'https://raw.githubusercontent.com/horovod/horovod/{hash}/examples/pytorch'  # noqa: E501
+        git_src = 'pytorch_synthetic_benchmark.py'
+        self.prerun_cmds = [
+            f'wget {git_url}/{git_src}',
+            f'echo starttime=`date +%s`'
```

Comment:
This this thing with the echos makes the time measurement a bit awkward. Couldn't you just `time` it? Also it seems that you are measuring also the creation of the virtual environment. Is this desired?

Reply:
thanks for the input @vkarak 
that elapsed time wasn't actually being monitored, so I've removed it completely now
(the venv should also go away in the upcomming version of the pytorch recipe)